packages/components/src/styles/components/app-side-nav/toggle-button.scss

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// APP-SIDE-NAV > TOGGLE BUTTON
+//
+
+.hds-app-side-nav__toggle-button {
```

Comment:
Bit of a nit: There's a small visual bug in the hover and active states of this collapse toggle button that appears to be related to some of those issues we were discussing with the svg clip path. Very hard and not really perceptible if you aren't paying attention.

Hover:
<img width="121" alt="Screenshot 2024-09-17 at 3 38 34 PM" src="https://github.com/user-attachments/assets/7b3143d1-8a43-4664-80c6-b80b1d4998a2">

Active:
<img width="111" alt="Screenshot 2024-09-17 at 3 38 46 PM" src="https://github.com/user-attachments/assets/fbaba4cd-abd8-434e-b36d-2110cafb8ad4">

Reply:
Would it be possible to not use a semi-transparent border color for the AppSideNav? That would fix issues like this without the need for fiddly CSS like the clip-path I added.

Reply:
@heatherlarsen Do you have any opinions/input you could share? There are issues with the semi-transparent border that affect the toggle tab design as shown. I came up with a partial mitigation but it's fiddly and there are remaining visual artifacts that are very difficult to fix as Jory screen-shotted above (I can keep experimenting though and will post an update here if I'm successful).

Reply:
I think for this scenario it makes sense to use a neutral color in place of the `Border/Primary` to solve this issue of transparency and overlap. In this case the closest is `Neutral/200`, I've already applied this to the component in [Figma](https://www.figma.com/design/noyY6dUMDYjmySpHcMjhkN/branch/WVcyo3y0R8Cf5ZatlfUkuk/HDS-Components?m=auto&node-id=59981-56567&t=NYJ9XGY2GiVrrpJZ-1) and it cleans up some of those problems already, so I'm good with making the slightly less systematic decision.

Reply:
Ok, I zoomed in and kept tweaking values in the browser until I got the clip paths looking better and figured out an issue with box-shadows that was also impacting rendering. I think I managed to come up with something that improves the rendering a good bit now but see what you think. @jorytindall

Reply:
I didn't see your comment above before I posted above, sorry. So should I just update the border color to use the solid color now then?

Reply:
Yeah, I think so, this is a bit more of a simple solution and is probably a bit more maintainable. Thanks for throwing that together though!

Reply:
Updated the border color in tokens and removed CSS which was no longer needed.